### PR TITLE
Fix JST identification

### PR DESCRIPTION
**Pull request by leeweisberger**
## Description

Fix what we identify as a JST in view params. There was a point when we were incorrectly storing JSTs in options as template objects. The views solution was developed during this time. JSTs are actually stored as strings, so update this logic to exactly match the logic in web: https://github.com/airplanedev/web/blob/1926779bdad15001c0dda1f96994df3af68d187b/components/parameters/ParameterInput.tsx#L627

Then I updated the template evaluation hook to echo back values that don't contain templates which required copying over the template parsing from web.

## Test plan
I added a test that ensures that a string template is evaluated and a string non-template is not.

### CODE REVIEW

File: lib/src/components/form/parameters.tsx [hunk 1]
```diff
@@ -492,10 +496,12 @@ const useTaskBackedConstraintOptions = ({
     : undefined;
 
   // Evaluate any templates that exist in the task params.
+  const canContainTemplate = (v: ParamValue): v is Template | string =>
```
Comment (leeweisberger) [inline lib/src/components/form/parameters.tsx:499]: This is copied from https://github.com/airplanedev/web/blob/1926779bdad15001c0dda1f96994df3af68d187b/components/parameters/ParameterInput.tsx#L627

This just handles both templates and strings that have templates. If the string has no template, useEvaluateTemplates just echos the value back

File: lib/src/components/form/parameters.tsx [hunk 2]
```diff
@@ -306,7 +307,10 @@ export const ParameterInput = ({
         defaultValue={canonicalizeValue(defaultValue, param.type)}
         data={options ?? []}
         loading={taskBackedConstraintLoading}
-        error={taskBackedConstraintError}
+        // Only include the taskBackedConstraintError if it is not empty as to not overwrite a validation error.
+        {...(taskBackedConstraintError
+          ? { error: taskBackedConstraintError }
+          : {})}
```
Comment (colinking) [inline lib/src/components/form/parameters.tsx:313]: Can you just do `error={taskBackedConstraintError ?? props.error}`?

File: lib/src/components/form/jst.ts [hunk 3]
```diff
@@ -3,6 +3,162 @@ import { useEffect, useRef, useState } from "react";
...
 import { Fetcher } from "client/fetcher";
 
+/**
+ * Parses a template string into template and non-template fragments. For example, the string
+ * `Hello {{ "world!"}}!` would be split into ["Hello ", `{{ "world!" }}`, "!"].
+ *
+ * A fragment in the returned response is a template if it starts with "{{".
+ *
+ * This logic is a direct port of `expressions.NewTemplate` from airport. Any changes here
+ * should be ported to airport and vice-versa.
```
Comment (colinking) [inline lib/src/components/form/jst.ts:13]: Can you update this comment (and in web) to indicate that it should also be kept up-to-date with `web`?

Comment (colinking) [reply lib/src/components/form/jst.ts:13]: Could it make sense to move this into `@airplane/lib`? We already import that in both places, so we could avoid duplication.

File: lib/src/components/form/jst.ts [hunk 4]
```diff
@@ -98,6 +254,12 @@ export const useEvaluateTemplates = (
         addResult(undefined, i);
       } else if (cacheRef.current[cacheKey]) {
         addResult(cacheRef.current[cacheKey], i);
+      } else if (
+        !opts.forceEvaluate &&
+        !newTemplate(unpackTemplate(t)).some((f) => f.startsWith("{{"))
+      ) {
+        // There is no template expression in this string, so we can just return it as-is.
+        addResult({ result: unpackTemplate(t) }, i);
```
Comment (colinking) [inline lib/src/components/form/jst.ts:262]: Can you execute a runbook from views?  Do we need to use `__airplaneType: "template"` objects in that case?

**Review by colinking — APPROVED** [review]

File: lib/src/components/form/parameters.tsx (hunk 2 above)
Comment (leeweisberger) [reply lib/src/components/form/parameters.tsx]: It's actually not `props.error` that is the problem. It's that error cannot be used in conjunction with the validate function which sets the error field downstream. I just fixed the root cause and reverted this

**Pull request merged**